app/models/bookmark.rb

```diff
@@ -1,3 +1,10 @@
+class ControllerValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    controllers = ["dashboard"] + (ActiveRecord::Base.connection.tables.map(&:to_s) + Permission.resources.map {|x| x.tableize }).uniq
```

Comment:
> Do you why have I removed https://github.com/theforeman/foreman/pull/2811/files#diff-a18735a12ab93f58a5ad67cf81a783ecR14 ?

No, I can't tell why you've removed it.  Is it because of what I guessed, or something else?



> I can revert that and invalidate on permission.create. I that what you mean?

It kind of was, but you could simply invalidate from the test rather than .create itself.  That is, if that's the reason you removed it.